src/rcdgen/nodes.cpp

```diff
@@ -974,6 +974,56 @@ int SHOPBlock::Write(FileWriter *fw)
 	return fw->AddBlock(fb);
 }
 
+FGTRBlock::FGTRBlock() : GameBlock("FGTR", 1)
+{
+}
+
+int FGTRBlock::Write(FileWriter *fw)
+{
+	FileBlock *fb = new FileBlock;
+	fb->StartSave(this->blk_name, this->version, 61 + 17 * (this->ride_width_x * this->ride_width_y) - 12);
+	fb->SaveUInt8(this->is_thrill_ride ? 1 : 0);
+	fb->SaveUInt16(this->tile_width);
+	fb->SaveUInt8(this->ride_width_x);
+	fb->SaveUInt8(this->ride_width_y);
+	for (int8 x = 0; x < this->ride_width_x; ++x) {
+		for (int8 y = 0; y < this->ride_width_y; ++y) {
+			fb->SaveUInt8(this->heights[x * ride_width_y + y]);
```

Comment:
The RCD block description doesn't say what the major and minor indices are, which is useful if you have a ride of at least 2x2.